include/deal.II/fe/fe_values.h

```diff
@@ -1663,9 +1663,9 @@ class FEValuesBase : protected FEValuesData<dim,spacedim>,
    *
    * @dealiiRequiresUpdateFlags{update_values}
    */
-  template <class InputVector, typename number>
+  template <class InputVector>
   void get_function_values (const InputVector &fe_function,
-                            std::vector<number> &values) const;
+                            std::vector<typename InputVector::value_type> &values) const;
```

Comment:
I am also in favor of eliminating the extra template and deduce from the global vector/matrix. However, I think we really need to document this because it takes quite some experience with C++ to understand the linker error in case someone used different types in the two vectors (or in the local and global matrix for ConstraintMatrix). I used this several times in my code so I guess we could hit a few problems in user code...